halibot/halibot.py

```diff
@@ -17,11 +17,18 @@
 	sys.path.append(".")
 import logging
 
+class ObjectDict(dict):
+	def __getattr__(self, key):
+		if key == "modules":
+			return dict(filter(lambda x: isinstance(x[1], HalModule), self.items()))
+		elif key == "agents":
+			return dict(filter(lambda x: isinstance(x[1], HalAgent), self.items()))
+		return super.__getattr__(self, key)
```

Comment:
Definitely is. Thanks for pointing me to that decorator, that might come in handy in other places too.